highfidelity/hifi
# Selection Scripting Interface

Implements a "Selection Scripting Interface", which allows JS or C++ to group gameplay items (entities, avatars, overlays) into a "selection", which then may or may not be pushed to the Scene.

There are a few parts to this PR:
1. Remove the old method of drawing a bounding box around "highlighted" entities (which used entity properties to determine whether or not an entity should be highlighted)
2. Add a new Selection Scripting Interface: Allow JS and C++ to pass a list name and entity/avatar/overlay IDs to the interface, then act on them by pushing them to the Scene.
3. Add a new Job to the Main Scene rendering task to take what's in the `"contextOverlayHighlightList"` Render Selection and draw a bounding box around the render item.

Eventually, (blocked on 11160) we need to hook in the outline shader.

**Test Plan:**
1. Run Interface, then close it.
2. Open `Interface.json`. Add `"inspectionMode": true,` to that file.
3. Open Interface.
4. Open the scripting console. Paste this in: `Render.getConfig("RenderMainView.DrawSelectionBounds").enabled = true`
5. Spawn an entity from the Marketplace
6. Verify that hoving over the entity with your mouse or hand controller lasers causes a bounding box to appear around the entity. (the bounding box will be replaced by the outline shader _"soon"_)
